R/feature.align.R

```diff
@@ -1,20 +1,91 @@
 to_attach <- function(pick, number_of_samples, use = "sum") {
     strengths <- rep(0, number_of_samples)
     if (is.null(nrow(pick))) {
+        # this is very strange if it can ever happen
+        # maybe commas are missing? we want the same as below
+        # but also why if there are no rows...
         strengths[pick[6]] <- pick[5]
         return(c(pick[1], pick[2], pick[1], pick[1], strengths))
     } else {
         for (i in seq_along(strengths)) {
+            # select all areas from the same sample
+            areas <- pick[pick[, 6] == i, 5]
             if (use == "sum")
-                strengths[i] <- sum(pick[pick[, 6] == i, 5])
+                strengths[i] <- sum(areas)
             if (use == "median")
-                strengths[i] <- median(pick[pick[, 6] == i, 5])
+                strengths[i] <- median(areas)
+            # can be NA if pick does not contain any data from a sample
         }
+        # average of m/z, average of rt, min of rt, max of rt, sum/median of areas
         return(c(mean(pick[, 1]), mean(pick[, 2]), min(pick[, 1]),
                  max(pick[, 1]), strengths))
     }
 }
 
+
+create_output <- function(sample_grouped, number_of_samples, deviation) {
+    return(c(to_attach(sample_grouped, number_of_samples, use = "sum"),
+             to_attach(sample_grouped[, c(1, 2, 3, 4, 2, 6)], number_of_samples, use = "median"),
```

Comment:
I'm not sure if I understand the question. `to_attach` is used to compute the sum of areas per sample (first call) and retention times (second call) - that's why in the second call, column `2` (RTs) is passed instead of column `5` (areas). I want to simplify this final step and also outputs of the alignment step as part of #87.